federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/HardcodedAttributeMapper.java

```diff
@@ -0,0 +1,228 @@
+        }
+    }
+
+    protected void checkDuplicateUsername(String userModelAttrName, String username, RealmModel realm,
+            KeycloakSession session, UserModel user) {
+        // only if working in USERNAME attribute
+        if (UserModel.USERNAME.equalsIgnoreCase(userModelAttrName)) {
+            if (username == null || username.isEmpty()) {
+                throw new ModelException("Cannot set an empty username");
+            }
+
+            boolean usernameChanged = !username.equals(user.getUsername());
+            if (realm.isEditUsernameAllowed() && usernameChanged) {
+                UserModel that = session.users().getUserByUsername(username, realm);
+                if (that != null && !that.getId().equals(user.getId())) {
+                    throw new ModelDuplicateException(String.format(
+                            "Cannot change the username to '%s' because the username already exists in keycloak",
+                            username), UserModel.USERNAME);
+                }
+            } else if (usernameChanged) {
+                throw new ModelException(
+                        "Cannot change username if the realm is not configured to allow edit the usernames");
+            }
+        }
+    }
+
+    @Override
+    public UserModel proxy(final LDAPObject ldapUser, UserModel delegate, RealmModel realm) {
+        String userModelAttrName = getUserModelAttribute();
+
+        String attributeValue = getAttributeValue();
+        Property<Object> userModelProperty = userModelProperties.get(userModelAttrName.toLowerCase());
+
+        delegate = new UserModelDelegate(delegate) {
+
+            @Override
+            public List<String> getAttribute(String name) {
+                if(userModelAttrName.equals(name)){
+                    return Arrays.asList(attributeValue);
+                }
+                return super.getAttribute(name);
+            }
+
+            @Override
+            public boolean isEmailVerified() {
+                if(userModelAttrName.equals("emailVerified")){
+                    return Boolean.valueOf(attributeValue);
+                }
+                return super.isEmailVerified();
+            }
+
+            @Override
+            public boolean isEnabled() {
+                if(userModelAttrName.equals("enabled")){
+                    return Boolean.valueOf(attributeValue);
+                }
+                return super.isEnabled();
+            }
+           
+       };
+       return delegate;
+   }
+
+   @Override
+   public void beforeLDAPQuery(LDAPQuery query) {
+       
+   }
+
+   private String getUserModelAttribute() {
+       return mapperModel.getConfig().getFirst(USER_MODEL_ATTRIBUTE);
+   }
+
+   String getLdapAttributeName() {
+       return mapperModel.getConfig().getFirst(LDAP_ATTRIBUTE);
+   }
+
+   String getAttributeValue() {
+      return mapperModel.getConfig().getFirst(ATTRIBUTE_VALUE);
+   }
+
+   private boolean isBinaryAttribute() {
```

Comment:
Same apply for those private methods, which are not used anymore. I understand you copied the implementation from some other class, but please cleanup unused stuff.